### PR TITLE
Add details to TimeProviders

### PR DESCRIPTION
**Pull request by j4sm1ne96**

### CODE REVIEW

**Review by rogersimmons — APPROVED** [review]

File: src/main/java/net/openhft/chronicle/core/time/SystemTimeProvider.java [hunk 1]
```diff
@@ -20,6 +20,10 @@
 
 import net.openhft.chronicle.core.Jvm;
 
+/**
+ * SystemTimeProvider synthesises an approximation via System.nanoTime deltas in combination with System.currentTimeMillis.
+ */
+
```
Comment (JerryShea) [inline src/main/java/net/openhft/chronicle/core/time/SystemTimeProvider.java]: pls remove blank line

File: src/main/java/net/openhft/chronicle/core/time/TimeProvider.java [hunk 2]
```diff
@@ -20,6 +20,14 @@
...
 import java.util.concurrent.TimeUnit;
 
+/**
+ * When looking at PosixTimeProvider vs SystemTimeProvider, PosixTimeProvider improves on SystemTimeProvider by calling
+ * directly into clock_gettime to get the nanosecond wall clock on platforms which support it.
+ * SystemTimeProvider can only synthesise an approximation via System.nanoTime deltas in combination with System.currentTimeMillis.
+ * <p>
+ * Between the two, PosixTimeProvider is more accurate and more stable than SystemTimeProvider, where there's a choice.
+ */
+
```
Comment (JerryShea) [inline src/main/java/net/openhft/chronicle/core/time/TimeProvider.java]: pls remove blank line

File: src/main/java/net/openhft/chronicle/core/time/PosixTimeProvider.java [hunk 3]
```diff
@@ -23,6 +23,8 @@
 
 /**
  * This is the raw Posix time, if you need it to be unique or a multiple of 32 use a wrapper.
+ * <p>
+ * PosixTimeProvider calls directly into clock_gettime to get the nanosecond wall clock on platforms which support it.
```
Comment (JerryShea) [inline src/main/java/net/openhft/chronicle/core/time/PosixTimeProvider.java]: maybe add "(native code)" after clock_gettime to make it clear that this calls into native code

File: src/main/java/net/openhft/chronicle/core/time/SystemTimeProvider.java (hunk 1 above)
Comment (JerryShea) [reply src/main/java/net/openhft/chronicle/core/time/SystemTimeProvider.java]: pls replace "an approximation" with "a nanosecond wall-clock time"

File: src/main/java/net/openhft/chronicle/core/time/TimeProvider.java (hunk 2 above)
Comment (JerryShea) [reply src/main/java/net/openhft/chronicle/core/time/TimeProvider.java]: suggest
```
/**
 * Provides a nanosecond resolution wall-clock timestamp.
 * <p>
 * There are a number of implementations. The simplest and fastest are {@link PosixTimeProvider} and 
 * {@link SystemTimeProvider}, with {@link PosixTimeProvider} recommended as being faster, more accurate and 
 * more stable, although it does make use of native code.
 */

```

**Comment by j4sm1ne96** [comment]
@JerryShea - if this is now approved, could you also please merge the PR as I am not authorised to do so

**Review by JerryShea — APPROVED** [review]

**Comment by JerryShea** [comment]
> @JerryShea - if this is now approved, could you also please merge the PR as I am not authorised to do so

thanks @j4sm1ne96 !

**Pull request merged**